python/mxnet/gluon/block.py

```diff
@@ -511,6 +517,9 @@ def _infer_attrs(self, infer_fn, attr, *args):
         args, _ = _flatten(args)
         arg_attrs, _, aux_attrs = getattr(out, infer_fn)(
             **{i.name: getattr(j, attr) for i, j in zip(inputs, args)})
+        if arg_attrs is None:
+            raise ValueError("Failed to obtain shape for %s"%(out))
```

Comment:
printing the output here doesn't help much. Look at the error message when symbol.bind cannot decide shape. It's has much more details